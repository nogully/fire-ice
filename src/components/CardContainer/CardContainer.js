import React, { Component } from 'react';
import { connect } from 'react-redux';
import { expandHouse } from '../../actions/actions'
import Card from '../Card/Card'

export class CardContainer extends Component {

  cardElements = () => {
    const { houses, expandHouse } = this.props;
    return houses.map((house, index) => {
      return <Card house={house} key={index} expandHouse={expandHouse} />
    })
  }
  
  render() {
    return (
      <div className="Container"> 
        { this.cardElements() } 
      </div>
    )
  }
}

export const mapStateToProps = ({ houses }) => ({ houses });

export const mapDispatchToProps = dispatch => ({ 
  expandHouse: house => dispatch(expandHouse(house))
});

export default connect(mapStateToProps, mapDispatchToProps)(CardContainer);
