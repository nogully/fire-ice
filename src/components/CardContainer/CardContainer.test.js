import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from './CardContainer'

describe('CardContainer', () => {
  const mockHouses = [
        { name: "House Dayne of Starfall"}, 
        { name: "House Lannister of Casterly Rock"}
      ]

  it('should look like its snapshot', () => {
    const wrapper = shallow(<CardContainer houses={mockHouses}  />)
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should define the value of houses prop for the CardContainer via Redux store', () => {
      const mockStore = { houses: [
              { name: "House Dayne of Starfall"}, 
              { name: "House Lannister of Casterly Rock"}
            ]}
      const expected = { houses: [
        { name: "House Dayne of Starfall"}, 
        { name: "House Lannister of Casterly Rock"}
      ]}
      const mapped = mapStateToProps(mockStore)
      expect(mapped).toEqual(expected)
    })
  })


})