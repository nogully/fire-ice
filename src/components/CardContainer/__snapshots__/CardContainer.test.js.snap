// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should look like its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    houses={
        Array [
            Object {
              "name": "House Dayne of Starfall",
            },
            Object {
              "name": "House Lannister of Casterly Rock",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          house={
                    Object {
                              "name": "House Dayne of Starfall",
                            }
          }
/>,
        <Unknown
          house={
                    Object {
                              "name": "House Lannister of Casterly Rock",
                            }
          }
/>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "house": Object {
            "name": "House Dayne of Starfall",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "house": Object {
            "name": "House Lannister of Casterly Rock",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            house={
                        Object {
                                    "name": "House Dayne of Starfall",
                                  }
            }
/>,
          <Unknown
            house={
                        Object {
                                    "name": "House Lannister of Casterly Rock",
                                  }
            }
/>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "house": Object {
              "name": "House Dayne of Starfall",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "house": Object {
              "name": "House Lannister of Casterly Rock",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
